### PR TITLE
Save recipes to Postgres

### PR DESCRIPTION
**Pull request by cb372**
* Introduce Quill to handle SQL stuff
* Use flyway-sbt for DB migrations
* Add a script to set up DB in dev environment
* Add a README

### CODE REVIEW

File: etl/src/main/scala/etl/ETL.scala [hunk 1]
```diff
@@ -120,3 +135,35 @@ object ETL extends App {
   }
 
 }
+
+object db {
+
+  lazy val ctx = new JdbcContext[PostgresDialect, SnakeCase]("db.ctx")
+  import ctx._
+
+  val a = quote(query[Recipe].insert)
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]: What is  'a' here? Can we have a more description variable name?

File: etl/src/main/scala/etl/ETL.scala [hunk 2]
```diff
@@ -120,3 +135,35 @@ object ETL extends App {
...
+
+  implicit val encodePublicationDate = mappedEncoding[OffsetDateTime, Date](d => Date.from(d.toInstant))
+  implicit val encodeStatus = mappedEncoding[Status, String](_.toString())
+
+  def jsonbEncoder[T: io.circe.Encoder: ClassTag] =
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]: Public methods should have a return type.

File: etl/src/main/scala/etl/ETL.scala [hunk 3]
```diff
@@ -120,3 +135,35 @@ object ETL extends App {
...
+      pgObj.setType("jsonb")
+      pgObj.setValue(value.asJson.noSpaces)
+      row.setObject(idx, pgObj, Types.OTHER)
+    }, Types.OTHER)
+
+  implicit val servesEncoder: Encoder[Serves] = jsonbEncoder[Serves]
+  implicit val stepsEncoder: Encoder[Steps] = jsonbEncoder[Steps]
+  implicit val ingredientsListsEncoder: Encoder[IngredientsLists] = jsonbEncoder[IngredientsLists]
+
+  def insertAll(recipes: Seq[Recipe]) {
```
Comment (LATaylor-guardian) [inline etl/src/main/scala/etl/ETL.scala]: As above.

**Comment by LATaylor-guardian** [comment]
👍

File: etl/src/main/scala/etl/ETL.scala (hunk 1 above)
Comment (tsop14) [inline etl/src/main/scala/etl/ETL.scala]: It's an action. Will change to just that.

**Pull request merged**